ipfs/library/Network/IPFS/Client/Files/Copy/Types.hs

```diff
@@ -0,0 +1,12 @@
+module Network.IPFS.Client.Files.Copy.Types (API) where
+
+import           Servant.API
+import           Network.IPFS.Prelude
+
+
+type API
+  = "cp"
+    :> QueryParam' '[Required] "arg" Text -- from
+    :> QueryParam' '[Required] "arg" Text -- to
```

Comment:
I had to double check against the HTTP API, because this is correct, but seems _like a terrible design._ Most/many libraries will return an array of `arg`s, and you depend on them being sent in the right order 😬 We may want to open an Issue on go-ipfs suggesting they use `from` and `to`.

Thoughts?

Reply:
My hunch is they have some sort of auto-generated CLI + HTTP API from some kind of schema, and if they changed it to be parameterized as `from` and `to` in the HTTP API, then their CLI API would have to look like this: `ipfs files cp --from=<path> --to=<path>`, which would be weird for someone used to `cp` in unix.
But maybe that's what `cp` should've been like from the start? :P

Reply:
Ah, indeed this is where they do that:

https://github.com/ipfs/kubo/blob/d82dec32482bb4ece33d3e867c4f615eea6bb487/core/corehttp/commands.go#L126-L152

Which is probably less work to maintain when you're doing all of this manually without much help from the compiler